trio/_core/_traps.py

```diff
@@ -64,7 +65,10 @@ class WaitTaskRescheduled:
     abort_func = attr.ib()
 
 
-async def wait_task_rescheduled(abort_func):
+RaiseCancelT = Callable[[], NoReturn]  # TypeAlias
+
+# Can this function be retyped to return something other than Any?
```

Comment:
Discussed: if you agree with my reasoning, then the answer seems to be no, not really.

Reply:
Feels like it might be possible to make `Task` generic and send around enough types that it all works out, but that's definitely out of scope for this PR.